libs/sdk/android/src/main/java/com/rudderstack/react/android/Utility.java

```diff
@@ -100,7 +104,7 @@ static RudderOption convertReadableMapToOptions(ReadableMap readableMap) {
         RudderOption options = new RudderOption();
```

Comment:
This whole function can be refactored for simplicity.

```
for (String key : optionsMap.keySet()) {
  if (key.equals(EXTERNAL_ID) || key.equals(EXTERNAL_IDS)) {
    //
  } else if (key.equals(INTEGRATIONS)) {
    // 
  } else {
    //
  }
}
```

You can also use a switch statement.

Reply:
There is one challenge:

Let me provide you with the previous context:

- Previously, we used to have the `externalIds` key in RN to set the external ID.
      - This is wrong behaviour, as in our document we mention setting `externalId`.
- Therefore in the past (refer to this [PR](https://github.com/rudderlabs/rudder-sdk-react-native/pull/155)), I corrected this behaviour in our RN code:
      - We started prioritising the `externalId` over `externalIds` i.e., if someone has both values in the `options` object in RN, we will only utilise the `externalId`.

Challenge:

- Now, if we move the setting of the `externalId` logic in the `switch` (or if-else) case. It’ll complicate things:
- This condition,  cannot guarantee to prefer `externalId`.
```
if (key.equals(EXTERNAL_ID) || key.equals(EXTERNAL_IDS))
```
- We would probably need some flags or a different workaround to implement this case. Which I think is not worth it.

So, I would request you please reconsider your suggestion and let me know if we should still do it (probably in a different way) or keep the current implementation as it is.

Reply:
Update: Based on our discussion on the call, we have reached the following optimizations:

- To simplify the logic for handling the options object, we had two options:
  - First, sort the options keys and handle them accordingly.
  - Second, remove the `externalIds` key when both `externalId` and `externalIds` are present, as this doesn't require any additional computation. We decided to proceed with this approach.

CC: @saikumarrs